src/main/java/net/openhft/chronicle/bytes/BytesStore.java

```diff
@@ -327,6 +328,11 @@ default boolean inside(@NonNegative long offset) {
 
     /**
      * Returns if a number of bytes starting from an offset are inside this ByteStore limits.
+     * If you are going to write n bytes starting at offset, you need to call this method with buffer=n-1,
+     * as the 1st byte is written at offset, and the last at offset+n-1
+     * <p>
+     * Use this test to determine if an offset is considered safe to write to. Note that it checks we are
+     * inside the BytesStore limits *including* the overlap
```

Comment:
I think the reason these errors exist is because this API is counter-intuitive. I reckon we should schedule a breaking change where the parameters for this method are interpreted as offset and length, meaning to check for a single byte you'd use the above method and to check for N = 2 or more bytes you'd call `inside(offset, N)`

Reply:
Although I appreciate that is an atrocious API change to make.

Reply:
It would be better to create a new method name and deprecate the old one.

Reply:
@JerryShea can we do this please? I really can see this being the source of a lot of subtle bugs going forward